### PR TITLE
[incubator/zookeeper] adding additional affinities

### PR DESCRIPTION
**Pull request by lnguyen**
*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

### CODE REVIEW

File: incubator/zookeeper/templates/statefulset.yaml [hunk 1]
```diff
@@ -16,8 +16,11 @@ spec:
         app: {{ include "zookeeper.name" . | quote }}
         release: {{ .Release.Name | quote }}
     spec:
-      {{- if eq .Values.antiAffinity "hard" }}
       affinity:
```
Comment (unguiculus) [inline incubator/zookeeper/templates/statefulset.yaml]: I don't like it to have `podAntiAffinity` fix in there plus additional affinities on top. I'd rather remove `podAntiAffinity` and provide full flexibility like so:

```yaml
{{- if .Values.affinity }}
      affinity:
{{ toYaml .Values.affinity | indent 8 }}
{{- end }}
```

There are other charts that do it like this.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/zookeeper/templates/statefulset.yaml (hunk 1 above)
Comment (lnguyen) [reply incubator/zookeeper/templates/statefulset.yaml]: I like that, I was trying to follow some of the others one. I'll switch it to this if its more of a standard

**Comment by lnguyen** [comment]
Updated

File: incubator/zookeeper/values.yaml [hunk 2]
```diff
@@ -26,9 +26,21 @@ snapRetain: 3
...
-antiAffinity: "hard"
 logLevel: "INFO"
 security:
   enabled: false
   runAsUser: 1000
   fsGroup: 1000
+## Affinities to add to the worker pods.
+## Useful if you prefer to run workers on non-spot instances, for example
+##
+# affinity:
```
Comment (unguiculus) [inline incubator/zookeeper/values.yaml]: Use this:

```yaml
  affinity: {}
  # ...
```

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by fedenusy** [comment]
@lachie83 @kow3ns if you want to keep the `soft` and `hard` options, please consider merging #3111 instead.

My vote goes towards this approach. Full flexibility is much better 👍 

If this gets merged I will close #3111 😄

**Comment by unguiculus** [comment]
Please fix the linting error. I think this line needs to be removed..

https://github.com/lnguyen/charts/blob/dd4de757739714ba377155387013ac2a2a85ff02/incubator/zookeeper/templates/statefulset.yaml#L23

**Pull request merged**